### PR TITLE
support fetching on multiple topics in one request

### PR DESCRIPTION
**Pull request by evanmcc**
OK, since so much had changed,  I had to spend some time reworking this one, but it should be all good to go now.  Roughly the same as before.

### CODE REVIEW

File: src/vg_client.erl [hunk 1]
```diff
@@ -42,39 +42,74 @@ ensure_topic(Topic) ->
...
+                  {ok, Pool} = vg_client_pool:get_pool(Topic, read),
+                  lager:debug("fetch request to pool: ~p ~p", [Topic, Pool]),
+                  case Acc of
+                      #{Pool := PoolReqs} ->
+                          Acc#{Pool => [R | PoolReqs]};
+                      _ ->
+                          Acc#{Pool => [R]}
+                  end
+          end, #{}, Requests),
+    %% should we do these in parallel?
```
Comment (tsloughter) [inline src/vg_client.erl:85]: Yea, maybe if `PoolReqs` is longer than 1 we use `ec_plists:map`?

File: src/vg_client.erl [hunk 2]
```diff
@@ -42,39 +42,74 @@ ensure_topic(Topic) ->
...
+                                 MaxBytes = maps:get(max_bytes, Opts, 0),
+                                 MaxIndex = maps:get(max_index, Opts, -1),
+                                 {Topic, [{0, Position, MaxBytes, MaxIndex}]}
+                             end
+                             || {Topic, Position, Opts} <- TPO0],
+                      case shackle:call(Pool, {fetch2, TPO}, Timeout) of
+                          {ok, Result} ->
+                              %% if there are any error codes in any
+                              %% of these, transform the whole thing
+                              %% into an error
```
Comment (tsloughter) [inline src/vg_client.erl:98]: I don't think that should be the case? The result should only be an error for the one of the requests that failed.

**Review by tsloughter — APPROVED** [review]

File: src/vg_client.erl (hunk 2 above)
Comment (evanmcc) [reply src/vg_client.erl:98]: I'm not really sure how to do this.  There are two possible error paths here, one via vonnegut error code, one via request failure (i.e. one of the requests in a multichain request set times out).  Should we fake that up into something that looks like an error that returned from vonnegut (i.e. assign error codes, etc)?

Comment (tsloughter) [reply src/vg_client.erl:98]: Not sure what you mean. But I may have misunderstood what this comment was for.

I'm just making sure that the results are considered separate. As in, the result to the user is a list of successful and failed responses, not having the fact that there is a failure in that list to change what is returned to the user.

**Pull request merged**